lib/R/Mbio/beta_div.R

```diff
@@ -0,0 +1,66 @@
+#   ## Inputs
+#   taxonomicLevel <- 'Class'
+#   k <- 20  # Keep pcoa dims
+#   method <- 'bray'
+#   verbose <- TRUE
+
+#   computeMessage <- ''
+  
+#   # Do argument checks
+#   # method = c('bray', 'jaccard', 'jsd')
+
+#   # Create otu table (samples x taxa)
+#   otu <- makeOTU(df, taxonomicLevel)
+#   plot.data::logWithTime(paste("Created OTU table with", NROW(otu), "samples and", (NCOL(otu)-1), "taxa."), verbose)
+
+#   # Compute beta div based on input
+#   if (identical(method, 'bray') | identical(method, 'jaccard')) {
+
+#     dist <- vegan::vegdist(otu[, -c('SampleID')], method=method, binary=TRUE)
+
+#   } else if (identical(method, 'jsd')) {
+
+#     # TO DO
+
+#   } else {
+#     stop('Unaccepted distance method. Accepted methods are bray, jaccard, and jsd.')
+#   }
+#   plot.data::logWithTime("Computed distance matrix.", verbose)
+
+#   # Ordination
+#   ## Need to handle how this might err
+#   pcoa <- ape::pcoa(dist)
+#   dt <- data.table::as.data.table(pcoa$vectors)
+#   computeMessage <- paste("PCoA returned results for", NCOL(dt), "dimensions.")
+#   dt$SampleID <- otu[['SampleID']]
+#   data.table::setcolorder(dt, c('SampleID'))
+#   plot.data::logWithTime("Finished ordination step.", verbose)
+
+#   # Extract percent variance
+#   eigenvecs <- pcoa$values$Relative_eig
+#   percentVar <- round(100*(eigenvecs / sum(eigenvecs)), 1)
+
+
+#   # Keep dims 1:20?
```

Comment:
idk yet what well want in the future. but id start by just returning whatever we mean to use to plot w, and change it if we need later.